From a91cbb33a9582ab384e5789f94284b74f3503085 Mon Sep 17 00:00:00 2001
From: Abdul Lateef Attar <abdattar@amd.com>
Date: Tue, 26 Oct 2021 16:58:21 +0530
Subject: [edk2-test] [PATCH v1 1/1] uefi-sct/SctPkg: Fix X64 build errors for GCC toolchain

REF: https://bugzilla.tianocore.org/show_bug.cgi?id=2126

* Correct function declarations/definitions by adding EFIAPI.
* Removes duplicate functions.
  SctRShiftU64 and SctDivU64x32 are already implemented in SctLib.
  MemoryFence is already implemented MdePkg/Library/BaseLib.
* Make mCrcTable in EfiCompliantBBTestSupport test static to avoid conflict
  with global sysmbol of the same name in BaseLib.

Signed-off-by: Abdul Lateef Attar <abdattar@amd.com>
Signed-off-by: Carolyn Gjertsen <Carolyn.Gjertsen@amd.com>
Tested-by: Heinrich Schuchrdt <heinrich.schuchardt@canonical.com>
---
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DebugSupport/BlackBoxTest/DebugSupportBBTestMain.h                  |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DevicePathFromText/BlackBoxTest/DevicePathFromTextBBTestMain.h      |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DevicePathToText/BlackBoxTest/DevicePathToTextBBTestMain.h          |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIConfigAccess/BlackBoxTest/HIIConfigAccessBBTestMain.h            |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIConfigRouting/BlackBoxTest/HIIConfigRoutingBBTestMain.h          |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIFontEx/BlackBoxTest/HIIFontExBBTestMain.h                        |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIImage/BlackBoxTest/HIIImageBBTestMain.h                          |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIImageEx/BlackBoxTest/HIIImageExBBTestMain.h                      |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIString/BlackBoxTest/HIIStringBBTestMain.h                        |  3 ++
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/LoadedImage/BlackBoxTest/LoadedImageBBTest.h                        |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/PartitionInfo/BlackBoxTest/PartitionInfoBBTestMain.h                |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/ResetNotification/BlackBoxTest/ResetNotificationBBTestMain.h        |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleFileSystem/BlackBoxTest/SimpleFileSystemBBTest.h              |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextIn/BlackBoxTest/SimpleTextInBBTestMain.h                  |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_efi.h            |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_uefi.h           |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/UFSDeviceConfig/BlackBoxTest/UFSDeviceConfigBBTestMain.h            |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/DevicePath/BlackBoxTest/DevicePathBBTestMain.h                      |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/HIIConfigAccess/BlackBoxTest/HIIConfigAccessBBTestMain.h            |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleFileSystem/BlackBoxTest/SimpleFileSystemBBTest.h              |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextIn/BlackBoxTest/SimpleTextInBBTestMain.h                  |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextInputEx/BlackBoxTest/SimpleTextInputExBBTestMain.h        |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_uefi.h           |  1 +
 uefi-sct/SctPkg/SCRT/SCRTDriver/ia32/Debug.c                                                                   |  9 ------
 uefi-sct/SctPkg/SCRT/SCRTDriver/x64/Debug.c                                                                    |  9 ------
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/AddressResolutionProtocol/Arp/ArpENTSTest.c                             |  1 +
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/AddressResolutionProtocol/ArpServiceBinding/ArpServiceBindingENTSTest.c |  3 ++
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/Dhcp4/Dhcp4/Dhcp4ENTSTest.c                                             |  1 +
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/Dhcp4/Dhcp4SB/Dhcp4SBENTSTest.c                                         |  3 ++
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/Dhcp6/Dhcp6/Dhcp6ENTSTest.c                                             |  1 +
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/Dhcp6/Dhcp6SB/Dhcp6SBENTSTest.c                                         |  1 +
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/Http/Http/HttpENTSTest.c                                                |  1 +
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/Http/HttpServiceBinding/HttpServiceBindingENTSTest.c                    |  3 ++
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol4/Ip4/Ip4ENTSTest.c                                     |  1 +
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol4/Ip4Config/Ip4ConfigENTSTest.c                         |  3 ++
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol4/Ip4Config2/Ip4Config2ENTSTest.c                       |  1 +
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol4/Ip4ServiceBinding/Ip4ServiceBindingENTSTest.c         |  3 ++
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol6/Ip6/Ip6ENTSTest.c                                     |  3 ++
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol6/Ip6Config/Ip6ConfigENTSTest.c                         |  3 ++
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol6/Ip6ServiceBinding/Ip6ServiceBindingENTSTest.c         |  1 +
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/ManagedNetwork/MNP/MnpENTSTest.c                                        |  1 +
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/ManagedNetwork/MNPSB/MnpSBENTSTest.c                                    |  1 +
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/Mtftp4/Mtftp4/Mtftp4ENTSTest.c                                          |  3 ++
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/Mtftp4/Mtftp4ServiceBinding/Mtftp4ServiceBindingENTSTest.c              |  1 +
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/Mtftp6/Mtftp6/Mtftp6ENTSTest.c                                          |  3 ++
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/Mtftp6/Mtftp6ServiceBinding/Mtftp6ServiceBindingENTSTest.c              |  1 +
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/PXEBaseCode/PXEBaseCodeENTSTest.c                                       |  3 ++
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/SimpleNetwork/SimpleNetworkENTSTest.c                                   |  1 +
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/Tcp4/Tcp4/Tcp4ENTSTest.c                                                |  3 ++
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/Tcp4/Tcp4ServiceBinding/Tcp4ServiceBindingENTSTest.c                    |  1 +
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/Tcp6/Tcp6/Tcp6ENTSTest.c                                                |  3 ++
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/Tcp6/Tcp6ServiceBinding/Tcp6ServiceBindingENTSTest.c                    |  1 +
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/Udp4/Udp4/Udp4ENTSTest.c                                                |  3 ++
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/Udp4/Udp4ServiceBinding/Udp4ServiceBindingENTSTest.c                    |  1 +
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/Udp6/Udp6/Udp6ENTSTest.c                                                |  1 +
 uefi-sct/SctPkg/TestCase/RIVL/Protocol/Udp6/Udp6ServiceBinding/Udp6ServiceBindingENTSTest.c                    |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Generic/EfiCompliant/BlackBoxTest/EfiCompliantBBTestSupport_uefi.c           |  2 +-
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DebugSupport/BlackBoxTest/DebugSupportBBTestMain.c                  |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DevicePathFromText/BlackBoxTest/DevicePathFromTextBBTestMain.c      |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DevicePathToText/BlackBoxTest/DevicePathToTextBBTestMain.c          |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/Ebc/BlackBoxTest/EbcBBTestMain.c                                    |  3 ++
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIConfigAccess/BlackBoxTest/HIIConfigAccessBBTestMain.c            |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIConfigRouting/BlackBoxTest/HIIConfigRoutingBBTestMain.c          |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIFontEx/BlackBoxTest/HIIFontExBBTestMain.c                        |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIImage/BlackBoxTest/HIIImageBBTestMain.c                          |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIImageEx/BlackBoxTest/HIIImageExBBTestMain.c                      |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIString/BlackBoxTest/HIIStringBBTestConformance.c                 |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIString/BlackBoxTest/HIIStringBBTestMain.c                        |  2 ++
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/LoadedImage/BlackBoxTest/LoadedImageBBTestMain.c                    |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/PartitionInfo/BlackBoxTest/PartitionInfoBBTestMain.c                |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/ResetNotification/BlackBoxTest/ResetNotificationBBTestMain.c        |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleFileSystem/BlackBoxTest/SimpleFileSystemBBTestMain.c          |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextIn/BlackBoxTest/SimpleTextInBBTestMain.c                  |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_efi.c            |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_uefi.c           |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/UFSDeviceConfig/BlackBoxTest/UFSDeviceConfigBBTestMain.c            |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/DevicePath/BlackBoxTest/DevicePathBBTestMain.c                      |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/Ebc/BlackBoxTest/EbcBBTestMain.c                                    |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/HIIConfigAccess/BlackBoxTest/HIIConfigAccessBBTestMain.c            |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleFileSystem/BlackBoxTest/SimpleFileSystemBBTestMain.c          |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextIn/BlackBoxTest/SimpleTextInBBTestMain.c                  |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextInputEx/BlackBoxTest/SimpleTextInputExBBTestMain.c        |  1 +
 uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_uefi.c           |  1 +
 uefi-sct/SctPkg/TestInfrastructure/SCT/Framework/ENTS/EasLib/Ebc/EfiLibPlat.c                                  | 29 --------------------
 uefi-sct/SctPkg/TestInfrastructure/SCT/Framework/ENTS/EasLib/X64/EfiLibPlat.c                                  | 29 --------------------
 85 files changed, 112 insertions(+), 77 deletions(-)

diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DebugSupport/BlackBoxTest/DebugSupportBBTestMain.h b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DebugSupport/BlackBoxTest/DebugSupportBBTestMain.h
index a9934a269380..aba49457a737 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DebugSupport/BlackBoxTest/DebugSupportBBTestMain.h
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DebugSupport/BlackBoxTest/DebugSupportBBTestMain.h
@@ -39,6 +39,7 @@ Abstract:
 extern EFI_INSTRUCTION_SET_ARCHITECTURE PlatformIsa;
 
 EFI_STATUS
+EFIAPI
 InitializeBBTestDebugSupport (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DevicePathFromText/BlackBoxTest/DevicePathFromTextBBTestMain.h b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DevicePathFromText/BlackBoxTest/DevicePathFromTextBBTestMain.h
index 22799075fdab..dfdd561734dd 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DevicePathFromText/BlackBoxTest/DevicePathFromTextBBTestMain.h
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DevicePathFromText/BlackBoxTest/DevicePathFromTextBBTestMain.h
@@ -38,6 +38,7 @@ Abstract:
 #define DEVICE_PATH_PROTOCOL_TEST_REVISION  0x00010001
 
 EFI_STATUS
+EFIAPI
 InitializeBBTestDevicePathFromTextProtocol (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DevicePathToText/BlackBoxTest/DevicePathToTextBBTestMain.h b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DevicePathToText/BlackBoxTest/DevicePathToTextBBTestMain.h
index b3e1bf365053..211b2228bf1d 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DevicePathToText/BlackBoxTest/DevicePathToTextBBTestMain.h
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DevicePathToText/BlackBoxTest/DevicePathToTextBBTestMain.h
@@ -67,6 +67,7 @@ typedef struct {
 
 
 EFI_STATUS
+EFIAPI
 InitializeBBTestDevicePathToTextProtocol (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIConfigAccess/BlackBoxTest/HIIConfigAccessBBTestMain.h b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIConfigAccess/BlackBoxTest/HIIConfigAccessBBTestMain.h
index 0857df89d76c..bedad19211b8 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIConfigAccess/BlackBoxTest/HIIConfigAccessBBTestMain.h
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIConfigAccess/BlackBoxTest/HIIConfigAccessBBTestMain.h
@@ -61,6 +61,7 @@ Abstract:
 
 
 EFI_STATUS
+EFIAPI
 InitializeHIIConfigAccessBBTest (
   IN EFI_HANDLE                   ImageHandle,
   IN EFI_SYSTEM_TABLE             *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIConfigRouting/BlackBoxTest/HIIConfigRoutingBBTestMain.h b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIConfigRouting/BlackBoxTest/HIIConfigRoutingBBTestMain.h
index c6ca192246c6..22f7d26c49e0 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIConfigRouting/BlackBoxTest/HIIConfigRoutingBBTestMain.h
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIConfigRouting/BlackBoxTest/HIIConfigRoutingBBTestMain.h
@@ -40,6 +40,7 @@ Abstract:
 
 
 EFI_STATUS
+EFIAPI
 InitializeBBTestHIIConfigRouting (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIFontEx/BlackBoxTest/HIIFontExBBTestMain.h b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIFontEx/BlackBoxTest/HIIFontExBBTestMain.h
index 593f1ee0d468..fde8e70fa483 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIFontEx/BlackBoxTest/HIIFontExBBTestMain.h
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIFontEx/BlackBoxTest/HIIFontExBBTestMain.h
@@ -86,6 +86,7 @@ Abstract:
 //
 //
 EFI_STATUS
+EFIAPI
 InitializeHIIFontExBBTest (
   IN EFI_HANDLE                   ImageHandle,
   IN EFI_SYSTEM_TABLE             *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIImage/BlackBoxTest/HIIImageBBTestMain.h b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIImage/BlackBoxTest/HIIImageBBTestMain.h
index 8bfcd36e281a..88a545c043ee 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIImage/BlackBoxTest/HIIImageBBTestMain.h
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIImage/BlackBoxTest/HIIImageBBTestMain.h
@@ -79,6 +79,7 @@ Abstract:
 //
 //
 EFI_STATUS
+EFIAPI
 InitializeHIIImageBBTest (
   IN EFI_HANDLE                   ImageHandle,
   IN EFI_SYSTEM_TABLE             *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIImageEx/BlackBoxTest/HIIImageExBBTestMain.h b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIImageEx/BlackBoxTest/HIIImageExBBTestMain.h
index 03deec936e37..f7bc1a0c548d 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIImageEx/BlackBoxTest/HIIImageExBBTestMain.h
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIImageEx/BlackBoxTest/HIIImageExBBTestMain.h
@@ -85,6 +85,7 @@ Abstract:
 //
 //
 EFI_STATUS
+EFIAPI
 InitializeHIIImageExBBTest (
   IN EFI_HANDLE                   ImageHandle,
   IN EFI_SYSTEM_TABLE             *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIString/BlackBoxTest/HIIStringBBTestMain.h b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIString/BlackBoxTest/HIIStringBBTestMain.h
index e062eba9fc5b..81be489bd868 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIString/BlackBoxTest/HIIStringBBTestMain.h
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIString/BlackBoxTest/HIIStringBBTestMain.h
@@ -77,6 +77,7 @@ Abstract:
 //
 //
 EFI_STATUS
+EFIAPI
 InitializeHIIStringBBTest (
   IN EFI_HANDLE                   ImageHandle,
   IN EFI_SYSTEM_TABLE             *SystemTable
@@ -84,6 +85,7 @@ InitializeHIIStringBBTest (
   
   
 EFI_STATUS
+EFIAPI
 UnloadHIIStringBBTest (
   IN EFI_HANDLE                   ImageHandle
   );
@@ -93,6 +95,7 @@ UnloadHIIStringBBTest (
 //Test Cases
 //
 EFI_STATUS
+EFIAPI
 BBTestNewStringConformanceTest (
   IN EFI_BB_TEST_PROTOCOL       *This,
   IN VOID                       *ClientInterface,
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/LoadedImage/BlackBoxTest/LoadedImageBBTest.h b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/LoadedImage/BlackBoxTest/LoadedImageBBTest.h
index eb3061ac14a5..1ad3e9a39c93 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/LoadedImage/BlackBoxTest/LoadedImageBBTest.h
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/LoadedImage/BlackBoxTest/LoadedImageBBTest.h
@@ -73,6 +73,7 @@ typedef struct {
 // Functions definitions
 //
 EFI_STATUS
+EFIAPI
 InitializeBBTestLoadedImageProtocol (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/PartitionInfo/BlackBoxTest/PartitionInfoBBTestMain.h b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/PartitionInfo/BlackBoxTest/PartitionInfoBBTestMain.h
index 93811a715ce5..33818f867436 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/PartitionInfo/BlackBoxTest/PartitionInfoBBTestMain.h
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/PartitionInfo/BlackBoxTest/PartitionInfoBBTestMain.h
@@ -46,6 +46,7 @@ Abstract:
 //
 //
 EFI_STATUS
+EFIAPI
 InitializePartitionInfoBBTest (
   IN EFI_HANDLE                   ImageHandle,
   IN EFI_SYSTEM_TABLE             *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/ResetNotification/BlackBoxTest/ResetNotificationBBTestMain.h b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/ResetNotification/BlackBoxTest/ResetNotificationBBTestMain.h
index f47e1ecc5369..84fb5ef9b8c7 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/ResetNotification/BlackBoxTest/ResetNotificationBBTestMain.h
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/ResetNotification/BlackBoxTest/ResetNotificationBBTestMain.h
@@ -71,6 +71,7 @@ extern UINT32      Attributes;
 //
 //
 EFI_STATUS
+EFIAPI
 InitializeResetNotificationBBTest (
   IN EFI_HANDLE                   ImageHandle,
   IN EFI_SYSTEM_TABLE             *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleFileSystem/BlackBoxTest/SimpleFileSystemBBTest.h b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleFileSystem/BlackBoxTest/SimpleFileSystemBBTest.h
index 4bb2c8e19da1..713d8b64dc56 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleFileSystem/BlackBoxTest/SimpleFileSystemBBTest.h
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleFileSystem/BlackBoxTest/SimpleFileSystemBBTest.h
@@ -229,6 +229,7 @@ typedef struct {
 //
 
 EFI_STATUS
+EFIAPI
 InitializeBBTestSimpleFileSystem (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextIn/BlackBoxTest/SimpleTextInBBTestMain.h b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextIn/BlackBoxTest/SimpleTextInBBTestMain.h
index ebb782212535..3497ba21e77f 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextIn/BlackBoxTest/SimpleTextInBBTestMain.h
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextIn/BlackBoxTest/SimpleTextInBBTestMain.h
@@ -61,6 +61,7 @@ extern   UINT8                  UnprintableKeyCheckingMap[][6];
 extern   CHAR16                 *UnprintableKeySymbol[];
 
 EFI_STATUS
+EFIAPI
 InitializeBBTestSimpleTextIn (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_efi.h b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_efi.h
index 75a0acb7efca..d515ea4d8f7c 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_efi.h
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_efi.h
@@ -46,6 +46,7 @@ extern   CHAR16                  *mAttribStr[];
 #define SIMPLE_TEXT_OUTPUT_PROTOCOL_TEST_REVISION 0x00010001
 
 EFI_STATUS
+EFIAPI
 InitializeBBTestSimpleTextOutput (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_uefi.h b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_uefi.h
index 882fbf691ebc..3dce653f22db 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_uefi.h
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_uefi.h
@@ -42,6 +42,7 @@ extern   CHAR16                  *mAttribStr[];
 #define SIMPLE_TEXT_OUTPUT_PROTOCOL_TEST_REVISION 0x00010001
 
 EFI_STATUS
+EFIAPI
 InitializeBBTestSimpleTextOutput (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/UFSDeviceConfig/BlackBoxTest/UFSDeviceConfigBBTestMain.h b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/UFSDeviceConfig/BlackBoxTest/UFSDeviceConfigBBTestMain.h
index 61c32ce43f28..5a7fbc5a75bf 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/UFSDeviceConfig/BlackBoxTest/UFSDeviceConfigBBTestMain.h
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/UFSDeviceConfig/BlackBoxTest/UFSDeviceConfigBBTestMain.h
@@ -52,6 +52,7 @@ Abstract:
 //
 //
 EFI_STATUS
+EFIAPI
 InitializeUFSDeviceConfigBBTest (
   IN EFI_HANDLE                   ImageHandle,
   IN EFI_SYSTEM_TABLE             *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/DevicePath/BlackBoxTest/DevicePathBBTestMain.h b/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/DevicePath/BlackBoxTest/DevicePathBBTestMain.h
index d2c422259eb3..691113f2868e 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/DevicePath/BlackBoxTest/DevicePathBBTestMain.h
+++ b/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/DevicePath/BlackBoxTest/DevicePathBBTestMain.h
@@ -39,6 +39,7 @@ Abstract:
   { 0xac7b0fd0, 0x884d, 0x45a8, 0x9d, 0x90, 0x99, 0x2a, 0x41, 0xa7, 0x88, 0x09 }
 
 EFI_STATUS
+EFIAPI
 InitializeBBTestDevicePath (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/HIIConfigAccess/BlackBoxTest/HIIConfigAccessBBTestMain.h b/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/HIIConfigAccess/BlackBoxTest/HIIConfigAccessBBTestMain.h
index acd0fabc7fc7..d2979d1cf214 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/HIIConfigAccess/BlackBoxTest/HIIConfigAccessBBTestMain.h
+++ b/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/HIIConfigAccess/BlackBoxTest/HIIConfigAccessBBTestMain.h
@@ -63,6 +63,7 @@ Abstract:
 
 
 EFI_STATUS
+EFIAPI
 InitializeHIIConfigAccessBBTest (
   IN EFI_HANDLE                   ImageHandle,
   IN EFI_SYSTEM_TABLE             *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleFileSystem/BlackBoxTest/SimpleFileSystemBBTest.h b/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleFileSystem/BlackBoxTest/SimpleFileSystemBBTest.h
index e869b52dd3a3..76ce3155e502 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleFileSystem/BlackBoxTest/SimpleFileSystemBBTest.h
+++ b/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleFileSystem/BlackBoxTest/SimpleFileSystemBBTest.h
@@ -233,6 +233,7 @@ typedef struct {
 //
 
 EFI_STATUS
+EFIAPI
 InitializeBBTestSimpleFileSystem (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextIn/BlackBoxTest/SimpleTextInBBTestMain.h b/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextIn/BlackBoxTest/SimpleTextInBBTestMain.h
index f03963ac8fa2..fe1081f73ac1 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextIn/BlackBoxTest/SimpleTextInBBTestMain.h
+++ b/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextIn/BlackBoxTest/SimpleTextInBBTestMain.h
@@ -64,6 +64,7 @@ extern   UINT8                  UnprintableKeyCheckingMap[][6];
 extern   CHAR16                 *UnprintableKeySymbol[];
 
 EFI_STATUS
+EFIAPI
 InitializeBBTestSimpleTextIn (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextInputEx/BlackBoxTest/SimpleTextInputExBBTestMain.h b/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextInputEx/BlackBoxTest/SimpleTextInputExBBTestMain.h
index bb0d8c9aef3a..88c13e2179d8 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextInputEx/BlackBoxTest/SimpleTextInputExBBTestMain.h
+++ b/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextInputEx/BlackBoxTest/SimpleTextInputExBBTestMain.h
@@ -92,6 +92,7 @@ typedef enum {
 //
 //
 EFI_STATUS
+EFIAPI
 InitializeSimpleTextInputExBBTest (
   IN EFI_HANDLE                   ImageHandle,
   IN EFI_SYSTEM_TABLE             *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_uefi.h b/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_uefi.h
index 59bc848825d4..d0854a1faac0 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_uefi.h
+++ b/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_uefi.h
@@ -46,6 +46,7 @@ extern   CHAR16                  *mAttribStr[];
   { 0x3395ddfa, 0xb843, 0x4fb2, 0xbc, 0x81, 0x10, 0xab, 0x95, 0xff, 0x87, 0x34 }
 
 EFI_STATUS
+EFIAPI
 InitializeBBTestSimpleTextOutput (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
diff --git a/uefi-sct/SctPkg/SCRT/SCRTDriver/ia32/Debug.c b/uefi-sct/SctPkg/SCRT/SCRTDriver/ia32/Debug.c
index fce42739f4db..76146b0dd2f5 100644
--- a/uefi-sct/SctPkg/SCRT/SCRTDriver/ia32/Debug.c
+++ b/uefi-sct/SctPkg/SCRT/SCRTDriver/ia32/Debug.c
@@ -26,15 +26,6 @@ Module Name:
 UINTN                 mHandOffPtr        =  0;
 EFI_PHYSICAL_ADDRESS  mIoPortSpaceAddress = 0;
 
-VOID
-EFIAPI
-MemoryFence (
-  VOID
-  )
-{
-  //TODO: Please implement me!
-}
-
 VOID
 UartReadWrite (
   IN BOOLEAN                 ReadFlag,
diff --git a/uefi-sct/SctPkg/SCRT/SCRTDriver/x64/Debug.c b/uefi-sct/SctPkg/SCRT/SCRTDriver/x64/Debug.c
index d2be4d3f479a..69a08f43e8da 100644
--- a/uefi-sct/SctPkg/SCRT/SCRTDriver/x64/Debug.c
+++ b/uefi-sct/SctPkg/SCRT/SCRTDriver/x64/Debug.c
@@ -25,15 +25,6 @@ Module Name:
 UINTN                 mHandOffPtr        =  0;
 EFI_PHYSICAL_ADDRESS  mIoPortSpaceAddress = 0;
 
-VOID
-EFIAPI
-MemoryFence (
-  VOID
-  )
-{
-  //TODO: Please implement me!
-}
-
 VOID
 UartReadWrite (
   IN BOOLEAN                 ReadFlag,
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/AddressResolutionProtocol/Arp/ArpENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/AddressResolutionProtocol/Arp/ArpENTSTest.c
index 62b64a4e15d7..93bd85c77dce 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/AddressResolutionProtocol/Arp/ArpENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/AddressResolutionProtocol/Arp/ArpENTSTest.c
@@ -77,6 +77,7 @@ ArpENTSTestUnload (
 
 
 EFI_STATUS
+EFIAPI
 ArpENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/AddressResolutionProtocol/ArpServiceBinding/ArpServiceBindingENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/AddressResolutionProtocol/ArpServiceBinding/ArpServiceBindingENTSTest.c
index 59ff5e544ba6..2e605c18054b 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/AddressResolutionProtocol/ArpServiceBinding/ArpServiceBindingENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/AddressResolutionProtocol/ArpServiceBinding/ArpServiceBindingENTSTest.c
@@ -46,12 +46,14 @@ ENTS_INTERFACE    gArpServiceBindingEntsInterfaceList[] = {
 EFI_ENTS_PROTOCOL *gArpServiceBindingEntsProtocolInterface = NULL;
 
 EFI_STATUS
+EFIAPI
 ArpServiceBindingENTSTestUnload (
   IN EFI_HANDLE                ImageHandle
   );
 
 
 EFI_STATUS
+EFIAPI
 ArpServiceBindingENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
@@ -136,6 +138,7 @@ Error:
 }
 
 EFI_STATUS
+EFIAPI
 ArpServiceBindingENTSTestUnload (
   IN EFI_HANDLE                ImageHandle
   )
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Dhcp4/Dhcp4/Dhcp4ENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Dhcp4/Dhcp4/Dhcp4ENTSTest.c
index 3747e1c2fb3d..b817775d8dd0 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Dhcp4/Dhcp4/Dhcp4ENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Dhcp4/Dhcp4/Dhcp4ENTSTest.c
@@ -87,6 +87,7 @@ Dhcp4ENTSTestUnload (
 
 
 EFI_STATUS
+EFIAPI
 Dhcp4ENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Dhcp4/Dhcp4SB/Dhcp4SBENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Dhcp4/Dhcp4SB/Dhcp4SBENTSTest.c
index 2b2af202fe03..774ee853bfca 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Dhcp4/Dhcp4SB/Dhcp4SBENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Dhcp4/Dhcp4SB/Dhcp4SBENTSTest.c
@@ -46,12 +46,14 @@ ENTS_INTERFACE    gDhcp4ServiceBindingEntsInterfaceList[] = {
 EFI_ENTS_PROTOCOL *gDhcp4ServiceBindingEntsProtocolInterface = NULL;
 
 EFI_STATUS
+EFIAPI
 Dhcp4SBENTSTestUnload (
   IN EFI_HANDLE                ImageHandle
   );
 
 
 EFI_STATUS
+EFIAPI
 Dhcp4SBENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
@@ -136,6 +138,7 @@ Error:
 }
 
 EFI_STATUS
+EFIAPI
 Dhcp4SBENTSTestUnload (
   IN EFI_HANDLE                ImageHandle
   )
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Dhcp6/Dhcp6/Dhcp6ENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Dhcp6/Dhcp6/Dhcp6ENTSTest.c
index 9c075ef37cc6..e5afe3fd4f2f 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Dhcp6/Dhcp6/Dhcp6ENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Dhcp6/Dhcp6/Dhcp6ENTSTest.c
@@ -89,6 +89,7 @@ Dhcp6ENTSTestUnload (
 
 
 EFI_STATUS
+EFIAPI
 Dhcp6ENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Dhcp6/Dhcp6SB/Dhcp6SBENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Dhcp6/Dhcp6SB/Dhcp6SBENTSTest.c
index 018828c17a0c..64feb5cb0ead 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Dhcp6/Dhcp6SB/Dhcp6SBENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Dhcp6/Dhcp6SB/Dhcp6SBENTSTest.c
@@ -53,6 +53,7 @@ Dhcp6SBENTSTestUnload (
 
 
 EFI_STATUS
+EFIAPI
 Dhcp6SBENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Http/Http/HttpENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Http/Http/HttpENTSTest.c
index 0e65fc0eb3d4..f380df707fff 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Http/Http/HttpENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Http/Http/HttpENTSTest.c
@@ -71,6 +71,7 @@ HttpENTSTestUnload (
 
 
 EFI_STATUS
+EFIAPI
 HttpENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Http/HttpServiceBinding/HttpServiceBindingENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Http/HttpServiceBinding/HttpServiceBindingENTSTest.c
index 5a538ca78dfa..102acd1d4c7d 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Http/HttpServiceBinding/HttpServiceBindingENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Http/HttpServiceBinding/HttpServiceBindingENTSTest.c
@@ -46,12 +46,14 @@ ENTS_INTERFACE    gHttpServiceBindingEntsInterfaceList[] = {
 EFI_ENTS_PROTOCOL *gHttpServiceBindingEntsProtocolInterface = NULL;
 
 EFI_STATUS
+EFIAPI
 HttpServiceBindingENTSTestUnload (
   IN EFI_HANDLE                ImageHandle
   );
 
 
 EFI_STATUS
+EFIAPI
 HttpServiceBindingENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
@@ -136,6 +138,7 @@ Error:
 }
 
 EFI_STATUS
+EFIAPI
 HttpServiceBindingENTSTestUnload (
   IN EFI_HANDLE                ImageHandle
   )
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol4/Ip4/Ip4ENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol4/Ip4/Ip4ENTSTest.c
index 5580e25296fc..4853c2a641ba 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol4/Ip4/Ip4ENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol4/Ip4/Ip4ENTSTest.c
@@ -82,6 +82,7 @@ Ip4ENTSTestUnload (
 
 
 EFI_STATUS
+EFIAPI
 Ip4ENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol4/Ip4Config/Ip4ConfigENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol4/Ip4Config/Ip4ConfigENTSTest.c
index e94d865b9940..24db7b9ef2fc 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol4/Ip4Config/Ip4ConfigENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol4/Ip4Config/Ip4ConfigENTSTest.c
@@ -52,12 +52,14 @@ ENTS_INTERFACE    gIp4ConfigEntsInterfaceList[] = {
 EFI_ENTS_PROTOCOL *gIp4ConfigEntsProtocolInterface = NULL;
 
 EFI_STATUS
+EFIAPI
 Ip4ConfigENTSTestUnload (
   IN EFI_HANDLE                ImageHandle
   );
 
 
 EFI_STATUS
+EFIAPI
 Ip4ConfigENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
@@ -133,6 +135,7 @@ Error:
 }
 
 EFI_STATUS
+EFIAPI
 Ip4ConfigENTSTestUnload (
   IN EFI_HANDLE                ImageHandle
   )
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol4/Ip4Config2/Ip4Config2ENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol4/Ip4Config2/Ip4Config2ENTSTest.c
index 6cdb544c0759..d4c3ad07340a 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol4/Ip4Config2/Ip4Config2ENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol4/Ip4Config2/Ip4Config2ENTSTest.c
@@ -63,6 +63,7 @@ Ip4Config2ENTSTestUnload (
 
 
 EFI_STATUS
+EFIAPI
 Ip4Config2ENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol4/Ip4ServiceBinding/Ip4ServiceBindingENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol4/Ip4ServiceBinding/Ip4ServiceBindingENTSTest.c
index b8c3271d018a..dc7aa96ca5db 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol4/Ip4ServiceBinding/Ip4ServiceBindingENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol4/Ip4ServiceBinding/Ip4ServiceBindingENTSTest.c
@@ -46,12 +46,14 @@ ENTS_INTERFACE    gIp4ServiceBindingEntsInterfaceList[] = {
 EFI_ENTS_PROTOCOL *gIp4ServiceBindingEntsProtocolInterface = NULL;
 
 EFI_STATUS
+EFIAPI
 Ip4ServiceBindingENTSTestUnload (
   IN EFI_HANDLE                ImageHandle
   );
 
 
 EFI_STATUS
+EFIAPI
 Ip4ServiceBindingENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
@@ -136,6 +138,7 @@ Error:
 }
 
 EFI_STATUS
+EFIAPI
 Ip4ServiceBindingENTSTestUnload (
   IN EFI_HANDLE                ImageHandle
   )
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol6/Ip6/Ip6ENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol6/Ip6/Ip6ENTSTest.c
index 69f5e19d17a7..090ea5827721 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol6/Ip6/Ip6ENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol6/Ip6/Ip6ENTSTest.c
@@ -82,12 +82,14 @@ ENTS_INTERFACE    gIp6EntsInterfaceList[] = {
 EFI_ENTS_PROTOCOL *gIp6EntsProtocolInterface = NULL;
 
 EFI_STATUS
+EFIAPI
 Ip6ENTSTestUnload (
   IN EFI_HANDLE                ImageHandle
   );
 
 
 EFI_STATUS
+EFIAPI
 Ip6ENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
@@ -163,6 +165,7 @@ Error:
 }
 
 EFI_STATUS
+EFIAPI
 Ip6ENTSTestUnload (
   IN EFI_HANDLE                ImageHandle
   )
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol6/Ip6Config/Ip6ConfigENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol6/Ip6Config/Ip6ConfigENTSTest.c
index 4bc7e518a64f..046c28b642cd 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol6/Ip6Config/Ip6ConfigENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol6/Ip6Config/Ip6ConfigENTSTest.c
@@ -56,12 +56,14 @@ ENTS_INTERFACE    gIp6ConfigEntsInterfaceList[] = {
 EFI_ENTS_PROTOCOL *gIp6ConfigEntsProtocolInterface = NULL;
 
 EFI_STATUS
+EFIAPI
 Ip6ConfigENTSTestUnload (
   IN EFI_HANDLE                ImageHandle
   );
 
 
 EFI_STATUS
+EFIAPI
 Ip6ConfigENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
@@ -146,6 +148,7 @@ Error:
 }
 
 EFI_STATUS
+EFIAPI
 Ip6ConfigENTSTestUnload (
   IN EFI_HANDLE                ImageHandle
   )
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol6/Ip6ServiceBinding/Ip6ServiceBindingENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol6/Ip6ServiceBinding/Ip6ServiceBindingENTSTest.c
index d84348f0a83e..b847fb056164 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol6/Ip6ServiceBinding/Ip6ServiceBindingENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/InternetProtocol6/Ip6ServiceBinding/Ip6ServiceBindingENTSTest.c
@@ -54,6 +54,7 @@ Ip6ServiceBindingENTSTestUnload (
 
 
 EFI_STATUS
+EFIAPI
 Ip6ServiceBindingENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/ManagedNetwork/MNP/MnpENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/ManagedNetwork/MNP/MnpENTSTest.c
index 5066c3c7fc1d..2a248d94ccca 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/ManagedNetwork/MNP/MnpENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/ManagedNetwork/MNP/MnpENTSTest.c
@@ -97,6 +97,7 @@ MnpENTSTestUnload (
 
 
 EFI_STATUS
+EFIAPI
 MnpENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/ManagedNetwork/MNPSB/MnpSBENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/ManagedNetwork/MNPSB/MnpSBENTSTest.c
index fe7d8a290e42..3e842786e2fb 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/ManagedNetwork/MNPSB/MnpSBENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/ManagedNetwork/MNPSB/MnpSBENTSTest.c
@@ -52,6 +52,7 @@ MnpSBENTSTestUnload (
 
 
 EFI_STATUS
+EFIAPI
 MnpSBENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Mtftp4/Mtftp4/Mtftp4ENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Mtftp4/Mtftp4/Mtftp4ENTSTest.c
index 7f8625c08a52..ecc75d9be96e 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Mtftp4/Mtftp4/Mtftp4ENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Mtftp4/Mtftp4/Mtftp4ENTSTest.c
@@ -75,12 +75,14 @@ ENTS_INTERFACE    gMtftp4EntsInterfaceList[] = {
 EFI_ENTS_PROTOCOL *gMtftp4EntsProtocolInterface = NULL;
 
 EFI_STATUS
+EFIAPI
 Mtftp4ENTSTestUnload (
   IN EFI_HANDLE                ImageHandle
   );
 
 
 EFI_STATUS
+EFIAPI
 Mtftp4ENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
@@ -157,6 +159,7 @@ Error:
 }
 
 EFI_STATUS
+EFIAPI
 Mtftp4ENTSTestUnload (
   IN EFI_HANDLE                ImageHandle
   )
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Mtftp4/Mtftp4ServiceBinding/Mtftp4ServiceBindingENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Mtftp4/Mtftp4ServiceBinding/Mtftp4ServiceBindingENTSTest.c
index 4eaa41bfece7..a50dbd2cbee1 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Mtftp4/Mtftp4ServiceBinding/Mtftp4ServiceBindingENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Mtftp4/Mtftp4ServiceBinding/Mtftp4ServiceBindingENTSTest.c
@@ -52,6 +52,7 @@ Mtftp4ServiceBindingENTSTestUnload (
 
 
 EFI_STATUS
+EFIAPI
 Mtftp4ServiceBindingENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Mtftp6/Mtftp6/Mtftp6ENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Mtftp6/Mtftp6/Mtftp6ENTSTest.c
index 7e17957cfc77..10c501c3e34e 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Mtftp6/Mtftp6/Mtftp6ENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Mtftp6/Mtftp6/Mtftp6ENTSTest.c
@@ -76,12 +76,14 @@ ENTS_INTERFACE    gMtftp6EntsInterfaceList[] = {
 EFI_ENTS_PROTOCOL *gMtftp6EntsProtocolInterface = NULL;
 
 EFI_STATUS
+EFIAPI
 Mtftp6ENTSTestUnload (
   IN EFI_HANDLE                ImageHandle
   );
 
 
 EFI_STATUS
+EFIAPI
 Mtftp6ENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
@@ -158,6 +160,7 @@ Error:
 }
 
 EFI_STATUS
+EFIAPI
 Mtftp6ENTSTestUnload (
   IN EFI_HANDLE                ImageHandle
   )
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Mtftp6/Mtftp6ServiceBinding/Mtftp6ServiceBindingENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Mtftp6/Mtftp6ServiceBinding/Mtftp6ServiceBindingENTSTest.c
index b87a43ef7d1b..07a1ca0d8dc6 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Mtftp6/Mtftp6ServiceBinding/Mtftp6ServiceBindingENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Mtftp6/Mtftp6ServiceBinding/Mtftp6ServiceBindingENTSTest.c
@@ -53,6 +53,7 @@ Mtftp6ServiceBindingENTSTestUnload (
 
 
 EFI_STATUS
+EFIAPI
 Mtftp6ServiceBindingENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/PXEBaseCode/PXEBaseCodeENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/PXEBaseCode/PXEBaseCodeENTSTest.c
index 5fef0062ff50..d4f1855a6c1f 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/PXEBaseCode/PXEBaseCodeENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/PXEBaseCode/PXEBaseCodeENTSTest.c
@@ -110,12 +110,14 @@ ENTS_INTERFACE    gPXEBaseCodeEntsInterfaceList[] = {
 EFI_ENTS_PROTOCOL *gPXEBaseCodeEntsProtocolInterface = NULL;
 
 EFI_STATUS
+EFIAPI
 PXEBaseCodeENTSTestUnload (
   IN EFI_HANDLE                ImageHandle
   );
 
 
 EFI_STATUS
+EFIAPI
 PXEBaseCodeENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
@@ -191,6 +193,7 @@ Error:
 }
 
 EFI_STATUS
+EFIAPI
 PXEBaseCodeENTSTestUnload (
   IN EFI_HANDLE                ImageHandle
   )
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/SimpleNetwork/SimpleNetworkENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/SimpleNetwork/SimpleNetworkENTSTest.c
index e99a4a604a26..bbcdb951141c 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/SimpleNetwork/SimpleNetworkENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/SimpleNetwork/SimpleNetworkENTSTest.c
@@ -126,6 +126,7 @@ SimpleNetworkENTSTestUnload (
 
 
 EFI_STATUS
+EFIAPI
 SimpleNetworkENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Tcp4/Tcp4/Tcp4ENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Tcp4/Tcp4/Tcp4ENTSTest.c
index 261e9adbe20e..a4daf90b1b45 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Tcp4/Tcp4/Tcp4ENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Tcp4/Tcp4/Tcp4ENTSTest.c
@@ -85,12 +85,14 @@ ENTS_INTERFACE    gTcp4EntsInterfaceList[] = {
 EFI_ENTS_PROTOCOL *gTcp4EntsProtocolInterface = NULL;
 
 EFI_STATUS
+EFIAPI
 Tcp4ENTSTestUnload (
   IN EFI_HANDLE                ImageHandle
   );
 
 
 EFI_STATUS
+EFIAPI
 Tcp4ENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
@@ -167,6 +169,7 @@ Error:
 }
 
 EFI_STATUS
+EFIAPI
 Tcp4ENTSTestUnload (
   IN EFI_HANDLE                ImageHandle
   )
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Tcp4/Tcp4ServiceBinding/Tcp4ServiceBindingENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Tcp4/Tcp4ServiceBinding/Tcp4ServiceBindingENTSTest.c
index 15e6b1ccbbb0..b887ee64afeb 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Tcp4/Tcp4ServiceBinding/Tcp4ServiceBindingENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Tcp4/Tcp4ServiceBinding/Tcp4ServiceBindingENTSTest.c
@@ -52,6 +52,7 @@ Tcp4ServiceBindingENTSTestUnload (
 
 
 EFI_STATUS
+EFIAPI
 Tcp4ServiceBindingENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Tcp6/Tcp6/Tcp6ENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Tcp6/Tcp6/Tcp6ENTSTest.c
index 3d4d5a35698c..b965f0f33452 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Tcp6/Tcp6/Tcp6ENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Tcp6/Tcp6/Tcp6ENTSTest.c
@@ -81,12 +81,14 @@ ENTS_INTERFACE    gTcp6EntsInterfaceList[] = {
 EFI_ENTS_PROTOCOL *gTcp6EntsProtocolInterface = NULL;
 
 EFI_STATUS
+EFIAPI
 Tcp6ENTSTestUnload (
   IN EFI_HANDLE                ImageHandle
   );
 
 
 EFI_STATUS
+EFIAPI
 Tcp6ENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
@@ -163,6 +165,7 @@ Error:
 }
 
 EFI_STATUS
+EFIAPI
 Tcp6ENTSTestUnload (
   IN EFI_HANDLE                ImageHandle
   )
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Tcp6/Tcp6ServiceBinding/Tcp6ServiceBindingENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Tcp6/Tcp6ServiceBinding/Tcp6ServiceBindingENTSTest.c
index e35f6fc1dddd..6c49bd1d0a78 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Tcp6/Tcp6ServiceBinding/Tcp6ServiceBindingENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Tcp6/Tcp6ServiceBinding/Tcp6ServiceBindingENTSTest.c
@@ -53,6 +53,7 @@ Tcp6ServiceBindingENTSTestUnload (
 
 
 EFI_STATUS
+EFIAPI
 Tcp6ServiceBindingENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Udp4/Udp4/Udp4ENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Udp4/Udp4/Udp4ENTSTest.c
index ddf1909f46d7..b0d03c58ad0e 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Udp4/Udp4/Udp4ENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Udp4/Udp4/Udp4ENTSTest.c
@@ -80,12 +80,14 @@ ENTS_INTERFACE    gUdp4EntsInterfaceList[] = {
 EFI_ENTS_PROTOCOL *gUdp4EntsProtocolInterface = NULL;
 
 EFI_STATUS
+EFIAPI
 Udp4ENTSTestUnload (
   IN EFI_HANDLE                ImageHandle
   );
 
 
 EFI_STATUS
+EFIAPI
 Udp4ENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
@@ -162,6 +164,7 @@ Error:
 }
 
 EFI_STATUS
+EFIAPI
 Udp4ENTSTestUnload (
   IN EFI_HANDLE                ImageHandle
   )
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Udp4/Udp4ServiceBinding/Udp4ServiceBindingENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Udp4/Udp4ServiceBinding/Udp4ServiceBindingENTSTest.c
index b3f47ee696fb..e42d1dc8793e 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Udp4/Udp4ServiceBinding/Udp4ServiceBindingENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Udp4/Udp4ServiceBinding/Udp4ServiceBindingENTSTest.c
@@ -52,6 +52,7 @@ Udp4ServiceBindingENTSTestUnload (
 
 
 EFI_STATUS
+EFIAPI
 Udp4ServiceBindingENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Udp6/Udp6/Udp6ENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Udp6/Udp6/Udp6ENTSTest.c
index 2c4b61af03b5..a90862064fe8 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Udp6/Udp6/Udp6ENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Udp6/Udp6/Udp6ENTSTest.c
@@ -82,6 +82,7 @@ Udp6ENTSTestUnload (
 
 
 EFI_STATUS
+EFIAPI
 Udp6ENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Udp6/Udp6ServiceBinding/Udp6ServiceBindingENTSTest.c b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Udp6/Udp6ServiceBinding/Udp6ServiceBindingENTSTest.c
index 00a8039ddc98..292e66f66a00 100644
--- a/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Udp6/Udp6ServiceBinding/Udp6ServiceBindingENTSTest.c
+++ b/uefi-sct/SctPkg/TestCase/RIVL/Protocol/Udp6/Udp6ServiceBinding/Udp6ServiceBindingENTSTest.c
@@ -53,6 +53,7 @@ Udp6ServiceBindingENTSTestUnload (
 
 
 EFI_STATUS
+EFIAPI
 Udp6ServiceBindingENTSTestMain (
   IN EFI_HANDLE                ImageHandle,
   IN EFI_SYSTEM_TABLE          *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Generic/EfiCompliant/BlackBoxTest/EfiCompliantBBTestSupport_uefi.c b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Generic/EfiCompliant/BlackBoxTest/EfiCompliantBBTestSupport_uefi.c
index 6017f41c2ef3..9e1bc9e14ef3 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Generic/EfiCompliant/BlackBoxTest/EfiCompliantBBTestSupport_uefi.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Generic/EfiCompliant/BlackBoxTest/EfiCompliantBBTestSupport_uefi.c
@@ -31,7 +31,7 @@ Abstract:
 // Module variables
 //
 
-UINT32  mCrcTable[256] = {
+STATIC UINT32  mCrcTable[256] = {
   0x00000000,
   0x77073096,
   0xEE0E612C,
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DebugSupport/BlackBoxTest/DebugSupportBBTestMain.c b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DebugSupport/BlackBoxTest/DebugSupportBBTestMain.c
index 29202445800a..0e3cd62e8dc0 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DebugSupport/BlackBoxTest/DebugSupportBBTestMain.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DebugSupport/BlackBoxTest/DebugSupportBBTestMain.c
@@ -115,6 +115,7 @@ BBTestDebugSupportProtocolUnload (
  *  @return EFI_SUCCESS the driver is loaded successfully.
  */
 EFI_STATUS
+EFIAPI
 InitializeBBTestDebugSupport (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DevicePathFromText/BlackBoxTest/DevicePathFromTextBBTestMain.c b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DevicePathFromText/BlackBoxTest/DevicePathFromTextBBTestMain.c
index 4252ff3ab626..a21bb132a03c 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DevicePathFromText/BlackBoxTest/DevicePathFromTextBBTestMain.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DevicePathFromText/BlackBoxTest/DevicePathFromTextBBTestMain.c
@@ -114,6 +114,7 @@ BBTestDevicePathFromTextProtocolUnload (
  *  @return EFI_SUCCESS the driver is loaded successfully.
  */
 EFI_STATUS
+EFIAPI
 InitializeBBTestDevicePathFromTextProtocol (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DevicePathToText/BlackBoxTest/DevicePathToTextBBTestMain.c b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DevicePathToText/BlackBoxTest/DevicePathToTextBBTestMain.c
index fc7cbb8113a2..b2c741816d72 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DevicePathToText/BlackBoxTest/DevicePathToTextBBTestMain.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/DevicePathToText/BlackBoxTest/DevicePathToTextBBTestMain.c
@@ -106,6 +106,7 @@ BBTestDevicePathToTextProtocolUnload (
  *  @return EFI_SUCCESS the driver is loaded successfully.
  */
 EFI_STATUS
+EFIAPI
 InitializeBBTestDevicePathToTextProtocol (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/Ebc/BlackBoxTest/EbcBBTestMain.c b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/Ebc/BlackBoxTest/EbcBBTestMain.c
index 84f21f0ced7f..2f1cd712d129 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/Ebc/BlackBoxTest/EbcBBTestMain.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/Ebc/BlackBoxTest/EbcBBTestMain.c
@@ -132,6 +132,7 @@ EFI_BB_TEST_ENTRY_FIELD gBBTestEntryField[] = {
 EFI_BB_TEST_PROTOCOL *gBBTestProtocolInterface;
 
 EFI_STATUS
+EFIAPI
 BBTestEbcUnload (
   IN EFI_HANDLE       ImageHandle
   );
@@ -148,6 +149,7 @@ BBTestEbcUnload (
  *  @return EFI_SUCCESS the driver is loaded successfully.
  */
 EFI_STATUS
+EFIAPI
 InitializeBBTestEbcProtocol (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
@@ -218,6 +220,7 @@ InitializeBBTestEbcProtocol (
  *  @return EFI_SUCCESS unload successfully.
  */
 EFI_STATUS
+EFIAPI
 BBTestEbcUnload (
   IN EFI_HANDLE       ImageHandle
   )
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIConfigAccess/BlackBoxTest/HIIConfigAccessBBTestMain.c b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIConfigAccess/BlackBoxTest/HIIConfigAccessBBTestMain.c
index f9d2fa0d42ca..cfb5fb1d4c18 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIConfigAccess/BlackBoxTest/HIIConfigAccessBBTestMain.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIConfigAccess/BlackBoxTest/HIIConfigAccessBBTestMain.c
@@ -124,6 +124,7 @@ EFI_BB_TEST_ENTRY_FIELD gBBTestEntryField[] = {
 
 
 EFI_STATUS
+EFIAPI
 InitializeHIIConfigAccessBBTest (
   IN EFI_HANDLE                   ImageHandle,
   IN EFI_SYSTEM_TABLE             *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIConfigRouting/BlackBoxTest/HIIConfigRoutingBBTestMain.c b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIConfigRouting/BlackBoxTest/HIIConfigRoutingBBTestMain.c
index 22820a95126e..6949c0241be0 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIConfigRouting/BlackBoxTest/HIIConfigRoutingBBTestMain.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIConfigRouting/BlackBoxTest/HIIConfigRoutingBBTestMain.c
@@ -186,6 +186,7 @@ EFI_BB_TEST_ENTRY_FIELD gBBTestEntryField[] = {
  *  @return EFI_SUCCESS the driver is loaded successfully.
  */
 EFI_STATUS
+EFIAPI
 InitializeBBTestHIIConfigRouting (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIFontEx/BlackBoxTest/HIIFontExBBTestMain.c b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIFontEx/BlackBoxTest/HIIFontExBBTestMain.c
index afb3518bcf9e..82d67e0718dd 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIFontEx/BlackBoxTest/HIIFontExBBTestMain.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIFontEx/BlackBoxTest/HIIFontExBBTestMain.c
@@ -145,6 +145,7 @@ EFI_BB_TEST_ENTRY_FIELD gBBTestEntryField[] = {
 //
 
 EFI_STATUS
+EFIAPI
 InitializeHIIFontExBBTest (
   IN EFI_HANDLE                   ImageHandle,
   IN EFI_SYSTEM_TABLE             *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIImage/BlackBoxTest/HIIImageBBTestMain.c b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIImage/BlackBoxTest/HIIImageBBTestMain.c
index 622c57ca3ee3..4bb1f43a4c80 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIImage/BlackBoxTest/HIIImageBBTestMain.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIImage/BlackBoxTest/HIIImageBBTestMain.c
@@ -144,6 +144,7 @@ EFI_BB_TEST_ENTRY_FIELD gBBTestEntryField[] = {
 //
 
 EFI_STATUS
+EFIAPI
 InitializeHIIImageBBTest (
   IN EFI_HANDLE                   ImageHandle,
   IN EFI_SYSTEM_TABLE             *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIImageEx/BlackBoxTest/HIIImageExBBTestMain.c b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIImageEx/BlackBoxTest/HIIImageExBBTestMain.c
index 47a54e716b4d..8a860a89a318 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIImageEx/BlackBoxTest/HIIImageExBBTestMain.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIImageEx/BlackBoxTest/HIIImageExBBTestMain.c
@@ -162,6 +162,7 @@ EFI_BB_TEST_ENTRY_FIELD gBBTestEntryField[] = {
 //
 
 EFI_STATUS
+EFIAPI
 InitializeHIIImageExBBTest (
   IN EFI_HANDLE                   ImageHandle,
   IN EFI_SYSTEM_TABLE             *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIString/BlackBoxTest/HIIStringBBTestConformance.c b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIString/BlackBoxTest/HIIStringBBTestConformance.c
index 995ec2a09ec7..783524468a80 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIString/BlackBoxTest/HIIStringBBTestConformance.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIString/BlackBoxTest/HIIStringBBTestConformance.c
@@ -90,6 +90,7 @@ BBTestGetSecondaryLanguagesConformanceTestCheckpoint1 (
 //Test Cases
 //
 EFI_STATUS
+EFIAPI
 BBTestNewStringConformanceTest (
   IN EFI_BB_TEST_PROTOCOL       *This,
   IN VOID                       *ClientInterface,
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIString/BlackBoxTest/HIIStringBBTestMain.c b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIString/BlackBoxTest/HIIStringBBTestMain.c
index 807e88db40bc..ca01cd33c09c 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIString/BlackBoxTest/HIIStringBBTestMain.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/HIIString/BlackBoxTest/HIIStringBBTestMain.c
@@ -145,6 +145,7 @@ EFI_BB_TEST_ENTRY_FIELD gBBTestEntryField[] = {
 //
 
 EFI_STATUS
+EFIAPI
 InitializeHIIStringBBTest (
   IN EFI_HANDLE                   ImageHandle,
   IN EFI_SYSTEM_TABLE             *SystemTable
@@ -172,6 +173,7 @@ InitializeHIIStringBBTest (
 
 
 EFI_STATUS
+EFIAPI
 UnloadHIIStringBBTest (
   IN EFI_HANDLE                   ImageHandle
   )
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/LoadedImage/BlackBoxTest/LoadedImageBBTestMain.c b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/LoadedImage/BlackBoxTest/LoadedImageBBTestMain.c
index 4a46e6425a77..b3b5afd19a41 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/LoadedImage/BlackBoxTest/LoadedImageBBTestMain.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/LoadedImage/BlackBoxTest/LoadedImageBBTestMain.c
@@ -78,6 +78,7 @@ EFI_BB_TEST_PROTOCOL *gBBTestProtocolInterface;
  *  @return EFI_INVALID_PARAMETER: One of the parameters has an invalid value.
  */
 EFI_STATUS
+EFIAPI
 InitializeBBTestLoadedImageProtocol (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/PartitionInfo/BlackBoxTest/PartitionInfoBBTestMain.c b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/PartitionInfo/BlackBoxTest/PartitionInfoBBTestMain.c
index 684265e67298..fc49266fce29 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/PartitionInfo/BlackBoxTest/PartitionInfoBBTestMain.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/PartitionInfo/BlackBoxTest/PartitionInfoBBTestMain.c
@@ -66,6 +66,7 @@ EFI_BB_TEST_ENTRY_FIELD gBBTestEntryField[] = {
 
 
 EFI_STATUS
+EFIAPI
 InitializePartitionInfoBBTest (
   IN EFI_HANDLE                   ImageHandle,
   IN EFI_SYSTEM_TABLE             *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/ResetNotification/BlackBoxTest/ResetNotificationBBTestMain.c b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/ResetNotification/BlackBoxTest/ResetNotificationBBTestMain.c
index 83704a2fe051..9e0233e62cbe 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/ResetNotification/BlackBoxTest/ResetNotificationBBTestMain.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/ResetNotification/BlackBoxTest/ResetNotificationBBTestMain.c
@@ -98,6 +98,7 @@ EFI_BB_TEST_ENTRY_FIELD gBBTestEntryField[] = {
 
 
 EFI_STATUS
+EFIAPI
 InitializeResetNotificationBBTest (
   IN EFI_HANDLE                   ImageHandle,
   IN EFI_SYSTEM_TABLE             *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleFileSystem/BlackBoxTest/SimpleFileSystemBBTestMain.c b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleFileSystem/BlackBoxTest/SimpleFileSystemBBTestMain.c
index 725e76c6829e..1cc7f12d4c65 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleFileSystem/BlackBoxTest/SimpleFileSystemBBTestMain.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleFileSystem/BlackBoxTest/SimpleFileSystemBBTestMain.c
@@ -400,6 +400,7 @@ EFI_BB_TEST_PROTOCOL *gBBTestProtocolInterface;
  *  @return EFI_INVALID_PARAMETER: One of the parameters has an invalid value.
  */
 EFI_STATUS
+EFIAPI
 InitializeBBTestSimpleFileSystem (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextIn/BlackBoxTest/SimpleTextInBBTestMain.c b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextIn/BlackBoxTest/SimpleTextInBBTestMain.c
index b878791dc1db..fe847f8f98eb 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextIn/BlackBoxTest/SimpleTextInBBTestMain.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextIn/BlackBoxTest/SimpleTextInBBTestMain.c
@@ -142,6 +142,7 @@ BBTestSimpleTextInProtocolUnload (
  *  @return EFI_SUCCESS the driver is loaded successfully.
  */
 EFI_STATUS
+EFIAPI
 InitializeBBTestSimpleTextIn (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_efi.c b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_efi.c
index 9ed9363bd321..ca2c3a083324 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_efi.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_efi.c
@@ -249,6 +249,7 @@ BBTestSimpleTextOutputProtocolUnload (
  *  @return EFI_SUCCESS the driver is loaded successfully.
  */
 EFI_STATUS
+EFIAPI
 InitializeBBTestSimpleTextOutput (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_uefi.c b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_uefi.c
index cce2f3e1ff23..2419ed8bbc68 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_uefi.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_uefi.c
@@ -257,6 +257,7 @@ BBTestSimpleTextOutputProtocolUnload (
  *  @return EFI_SUCCESS the driver is loaded successfully.
  */
 EFI_STATUS
+EFIAPI
 InitializeBBTestSimpleTextOutput (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/UFSDeviceConfig/BlackBoxTest/UFSDeviceConfigBBTestMain.c b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/UFSDeviceConfig/BlackBoxTest/UFSDeviceConfigBBTestMain.c
index 04caf3ce5d6c..e3b96f857825 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/UFSDeviceConfig/BlackBoxTest/UFSDeviceConfigBBTestMain.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/UFSDeviceConfig/BlackBoxTest/UFSDeviceConfigBBTestMain.c
@@ -82,6 +82,7 @@ EFI_BB_TEST_ENTRY_FIELD gBBTestEntryField[] = {
 
 
 EFI_STATUS
+EFIAPI
 InitializeUFSDeviceConfigBBTest (
   IN EFI_HANDLE                   ImageHandle,
   IN EFI_SYSTEM_TABLE             *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/DevicePath/BlackBoxTest/DevicePathBBTestMain.c b/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/DevicePath/BlackBoxTest/DevicePathBBTestMain.c
index 2c0349e57363..30a394ef5142 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/DevicePath/BlackBoxTest/DevicePathBBTestMain.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/DevicePath/BlackBoxTest/DevicePathBBTestMain.c
@@ -69,6 +69,7 @@ BBTestDevicePathProtocolUnload (
  *  @return EFI_SUCCESS the driver is loaded successfully.
  */
 EFI_STATUS
+EFIAPI
 InitializeBBTestDevicePath (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/Ebc/BlackBoxTest/EbcBBTestMain.c b/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/Ebc/BlackBoxTest/EbcBBTestMain.c
index b7842575f74c..b5fec2a029f1 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/Ebc/BlackBoxTest/EbcBBTestMain.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/Ebc/BlackBoxTest/EbcBBTestMain.c
@@ -144,6 +144,7 @@ BBTestEbcUnload (
  *  @return EFI_SUCCESS the driver is loaded successfully.
  */
 EFI_STATUS
+EFIAPI
 InitializeBBTestEbcProtocol (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/HIIConfigAccess/BlackBoxTest/HIIConfigAccessBBTestMain.c b/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/HIIConfigAccess/BlackBoxTest/HIIConfigAccessBBTestMain.c
index 92de424a496f..4c5bb1e6a4c9 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/HIIConfigAccess/BlackBoxTest/HIIConfigAccessBBTestMain.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/HIIConfigAccess/BlackBoxTest/HIIConfigAccessBBTestMain.c
@@ -90,6 +90,7 @@ EFI_BB_TEST_ENTRY_FIELD gBBTestEntryField[] = {
 
 
 EFI_STATUS
+EFIAPI
 InitializeHIIConfigAccessBBTest (
   IN EFI_HANDLE                   ImageHandle,
   IN EFI_SYSTEM_TABLE             *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleFileSystem/BlackBoxTest/SimpleFileSystemBBTestMain.c b/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleFileSystem/BlackBoxTest/SimpleFileSystemBBTestMain.c
index 629f41d49736..6bfbfda9d27d 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleFileSystem/BlackBoxTest/SimpleFileSystemBBTestMain.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleFileSystem/BlackBoxTest/SimpleFileSystemBBTestMain.c
@@ -400,6 +400,7 @@ EFI_BB_TEST_PROTOCOL *gBBTestProtocolInterface;
  *  @return EFI_INVALID_PARAMETER: One of the parameters has an invalid value.
  */
 EFI_STATUS
+EFIAPI
 InitializeBBTestSimpleFileSystem (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextIn/BlackBoxTest/SimpleTextInBBTestMain.c b/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextIn/BlackBoxTest/SimpleTextInBBTestMain.c
index 38f94dc5024a..b66e21fad31d 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextIn/BlackBoxTest/SimpleTextInBBTestMain.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextIn/BlackBoxTest/SimpleTextInBBTestMain.c
@@ -142,6 +142,7 @@ BBTestSimpleTextInProtocolUnload (
  *  @return EFI_SUCCESS the driver is loaded successfully.
  */
 EFI_STATUS
+EFIAPI
 InitializeBBTestSimpleTextIn (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextInputEx/BlackBoxTest/SimpleTextInputExBBTestMain.c b/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextInputEx/BlackBoxTest/SimpleTextInputExBBTestMain.c
index c0a4f5aa51b5..29e09c6b07cb 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextInputEx/BlackBoxTest/SimpleTextInputExBBTestMain.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextInputEx/BlackBoxTest/SimpleTextInputExBBTestMain.c
@@ -160,6 +160,7 @@ BOOLEAN     BeenExecuted;
 //
 
 EFI_STATUS
+EFIAPI
 InitializeSimpleTextInputExBBTest (
   IN EFI_HANDLE                   ImageHandle,
   IN EFI_SYSTEM_TABLE             *SystemTable
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_uefi.c b/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_uefi.c
index e72826bbaaaf..0cb29973240c 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_uefi.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextOut/BlackBoxTest/SimpleTextOutBBTestMain_uefi.c
@@ -258,6 +258,7 @@ BBTestSimpleTextOutputProtocolUnload (
  *  @return EFI_SUCCESS the driver is loaded successfully.
  */
 EFI_STATUS
+EFIAPI
 InitializeBBTestSimpleTextOutput (
   IN EFI_HANDLE           ImageHandle,
   IN EFI_SYSTEM_TABLE     *SystemTable
diff --git a/uefi-sct/SctPkg/TestInfrastructure/SCT/Framework/ENTS/EasLib/Ebc/EfiLibPlat.c b/uefi-sct/SctPkg/TestInfrastructure/SCT/Framework/ENTS/EasLib/Ebc/EfiLibPlat.c
index c942823bb8e0..4fbf6a735912 100644
--- a/uefi-sct/SctPkg/TestInfrastructure/SCT/Framework/ENTS/EasLib/Ebc/EfiLibPlat.c
+++ b/uefi-sct/SctPkg/TestInfrastructure/SCT/Framework/ENTS/EasLib/Ebc/EfiLibPlat.c
@@ -27,35 +27,6 @@ Abstract:
 #include "Efi.h"
 #include "EntsLibPlat.h"
 
-UINT64
-SctRShiftU64 (
-  IN UINT64   Operand,
-  IN UINTN    Count
-  )
-{
-  return Operand >> Count;
-}
-
-UINT64
-SctDivU64x32 (
-  IN UINT64   Dividend,
-  IN UINTN    Divisor,
-  OUT UINTN   *Remainder OPTIONAL
-  )
-{
-  ASSERT (Divisor != 0);
-
-  //
-  // Have to compute twice if remainder. No support for
-  // divide-with-remainder in VM.
-  //
-  if (Remainder != NULL) {
-    *Remainder = Dividend % Divisor;
-  }
-
-  return Dividend / Divisor;
-}
-
 VOID
 EntsInitializeLibPlatform (
   IN EFI_HANDLE           ImageHandle,
diff --git a/uefi-sct/SctPkg/TestInfrastructure/SCT/Framework/ENTS/EasLib/X64/EfiLibPlat.c b/uefi-sct/SctPkg/TestInfrastructure/SCT/Framework/ENTS/EasLib/X64/EfiLibPlat.c
index c942823bb8e0..4fbf6a735912 100644
--- a/uefi-sct/SctPkg/TestInfrastructure/SCT/Framework/ENTS/EasLib/X64/EfiLibPlat.c
+++ b/uefi-sct/SctPkg/TestInfrastructure/SCT/Framework/ENTS/EasLib/X64/EfiLibPlat.c
@@ -27,35 +27,6 @@ Abstract:
 #include "Efi.h"
 #include "EntsLibPlat.h"
 
-UINT64
-SctRShiftU64 (
-  IN UINT64   Operand,
-  IN UINTN    Count
-  )
-{
-  return Operand >> Count;
-}
-
-UINT64
-SctDivU64x32 (
-  IN UINT64   Dividend,
-  IN UINTN    Divisor,
-  OUT UINTN   *Remainder OPTIONAL
-  )
-{
-  ASSERT (Divisor != 0);
-
-  //
-  // Have to compute twice if remainder. No support for
-  // divide-with-remainder in VM.
-  //
-  if (Remainder != NULL) {
-    *Remainder = Dividend % Divisor;
-  }
-
-  return Dividend / Divisor;
-}
-
 VOID
 EntsInitializeLibPlatform (
   IN EFI_HANDLE           ImageHandle,
-- 
2.25.1

